### ISSUE TITLE
Psst closes on login attempt

### ISSUE THREAD

**Issue by arsnm** [jpochyla/psst#551]
Hi,

On Windows, when trying to login to Spotify, the app closes just after loging in.

Steps to reproduce:
1. Open the psst-gui.exe
2. Click "Login with Spotify"
3. *On the browser* Click "Continue to the app"
4. After being redirected to the page saying "You can close the window now", psst-gui.exe closes itself

 - OS: Windows 11 23H2
 - Using PSST downloaded from the link provided for Windows in the "Download" section

Thanks in advance for the help.

**Comment by arsnm** [comment]
Additional Context : I tested the app on MacOS (using the same Spotify account and steps) and it does work, so I think the problem is Windows-related.

**Comment by lotalis** [comment]
I would like to +1 this, may not be an isolated issue.

**Comment by Montero1905** [comment]
+1

**Comment by ericmedina024** [comment]
> Are there any error messages when built on Windows from this?

Seems to be an issue with creating `%appdata%/Psst`. Creating this folder manually resolves the issue and allows Psst to start normally.

```
C:\Users\Eric\RustroverProjects\psst>cargo run --bin psst-gui
warning: error finalizing incremental compilation session directory `\\?\C:\Users\Eric\RustroverProjects\psst\target\debug\incremental\psst_protocol-268ihjyy4u29u\s-h1g26aime5-0otynen-working`: Access is denied. (os error 5)

warning: `psst-protocol` (lib) generated 1 warning
   Compiling psst-core v0.1.0 (C:\Users\Eric\RustroverProjects\psst\psst-core)
   Compiling psst-gui v0.1.0 (C:\Users\Eric\RustroverProjects\psst\psst-gui)
    Finished `dev` profile [optimized + debuginfo] target(s) in 5.21s
     Running `target\debug\psst-gui.exe`
[2024-11-03T01:57:26Z INFO  psst_core::oauth] Starting OAuth listener on 127.0.0.1:8888
[2024-11-03T01:57:26Z INFO  psst_core::oauth] Listener bound successfully
[2024-11-03T01:57:28Z ERROR psst_gui::webapi::client] failed to read local tracks: The system cannot find the path specified. (os error 3)
thread 'main' panicked at psst-gui\src\data\config.rs:165:35:
Failed to create config dir: Os { code: 3, kind: NotFound, message: "The system cannot find the path specified." }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at core\src\panicking.rs:221:5:
panic in a function that cannot unwind
stack backtrace:
   0:     0x7ff681febec1 - std::backtrace_rs::backtrace::dbghelp64::trace
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library/std\src\..\..\backtrace\src\backtrace\dbghelp64.rs:91
   1:     0x7ff681febec1 - std::backtrace_rs::backtrace::trace_unsynchronized
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library/std\src\..\..\backtrace\src\backtrace\mod.rs:66
   2:     0x7ff681febec1 - std::sys::backtrace::_print_fmt
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library/std\src\sys\backtrace.rs:66
   3:     0x7ff681febec1 - std::sys::backtrace::impl$0::print::impl$0::fmt
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library/std\src\sys\backtrace.rs:39
   4:     0x7ff6820108c9 - core::fmt::rt::Argument::fmt
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library/core\src\fmt\rt.rs:177
   5:     0x7ff6820108c9 - core::fmt::write
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library/core\src\fmt\mod.rs:1178
   6:     0x7ff681fe6d07 - std::io::Write::write_fmt<std::sys::pal::windows::stdio::Stderr>
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library/std\src\io\mod.rs:1823
   7:     0x7ff681fedfd9 - std::panicking::default_hook::closure$1
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library/std\src\panicking.rs:266
   8:     0x7ff681fedb5c - std::panicking::default_hook
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library/std\src\panicking.rs:293
   9:     0x7ff681fee8e2 - std::panicking::rust_panic_with_hook
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library/std\src\panicking.rs:797
  10:     0x7ff681fee6ef - std::panicking::begin_panic_handler::closure$0
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library/std\src\panicking.rs:664
  11:     0x7ff681fec97f - std::sys::backtrace::__rust_end_short_backtrace<std::panicking::begin_panic_handler::closure_env$0,never$>
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library/std\src\sys\backtrace.rs:170
  12:     0x7ff681fee336 - std::panicking::begin_panic_handler
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library/std\src\panicking.rs:662
  13:     0x7ff68203f6dd - core::panicking::panic_nounwind_fmt::runtime
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library/core\src\panicking.rs:112
  14:     0x7ff68203f6dd - core::panicking::panic_nounwind_fmt
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library/core\src\panicking.rs:122
  15:     0x7ff68203f783 - core::panicking::panic_nounwind
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library/core\src\panicking.rs:221
  16:     0x7ff68203f93b - core::panicking::panic_cannot_unwind
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library/core\src\panicking.rs:310
  17:     0x7ff681f5745e - druid_shell::backend::windows::window::win_proc_dispatch
                               at C:\Users\Eric\.cargo\git\checkouts\druid-64a54200c8cd3ebf\230f7dd\druid-shell\src\backend\windows\window.rs:1974
  18:     0x7fffbb0bf540 - _CxxFrameHandler3
  19:     0x7fffbb0b33d8 - is_exception_typeof
  20:     0x7fffdca83916 - RtlCaptureContext2
  21:     0x7ff681f57372 - druid_shell::backend::windows::window::win_proc_dispatch
                               at C:\Users\Eric\.cargo\git\checkouts\druid-64a54200c8cd3ebf\230f7dd\druid-shell\src\backend\windows\window.rs:1990
  22:     0x7fffdc535801 - CallWindowProcW
  23:     0x7fffdc53334d - IsWindowUnicode
  24:     0x7ff681f2ee40 - druid_shell::application::Application::run
                               at C:\Users\Eric\.cargo\git\checkouts\druid-64a54200c8cd3ebf\230f7dd\druid-shell\src\application.rs:150
  25:     0x7ff680d12e7c - druid::app::AppLauncher<psst_gui::data::AppState>::launch<psst_gui::data::AppState>
                               at C:\Users\Eric\.cargo\git\checkouts\druid-64a54200c8cd3ebf\230f7dd\druid\src\app.rs:282
  26:     0x7ff6811a683c - psst_gui::main
                               at C:\Users\Eric\RustroverProjects\psst\psst-gui\src\main.rs:62
  27:     0x7ff6817605a6 - core::ops::function::FnOnce::call_once
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library\core\src\ops\function.rs:250
  28:     0x7ff6817605a6 - std::sys::backtrace::__rust_begin_short_backtrace<void (*)(),tuple$<> >
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library\std\src\sys\backtrace.rs:154
  29:     0x7ff680d8ec1c - std::rt::lang_start::closure$0
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library\std\src\rt.rs:164
  30:     0x7ff680d8ec1c - core::ops::function::FnOnce::call_once
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library\core\src\ops\function.rs:250
  31:     0x7ff680d8ec1c - core::ops::function::FnOnce::call_once<std::rt::lang_start::closure_env$0<tuple$<> >,tuple$<> >
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library\std\src\rt.rs:164
  32:     0x7ff681fdf159 - std::rt::lang_start_internal::closure$2
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library/std\src\rt.rs:143
  33:     0x7ff681fdf159 - std::panicking::try::do_call
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library/std\src\panicking.rs:554
  34:     0x7ff681fdf159 - std::panicking::try
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library/std\src\panicking.rs:518
  35:     0x7ff681fdf159 - std::panic::catch_unwind
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library/std\src\panic.rs:345
  36:     0x7ff681fdf159 - std::rt::lang_start_internal
                               at /rustc/f6e511eec7342f59a25f7c0534f1dbea00d01b14\library/std\src\rt.rs:143
  37:     0x7ff6811a6f2c - main
  38:     0x7ff682022e0c - invoke_main
                               at D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:78
  39:     0x7ff682022e0c - __scrt_common_main_seh
                               at D:\a\_work\1\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288
  40:     0x7fffdb13dbe7 - BaseThreadInitThunk
  41:     0x7fffdc9e5a4c - RtlUserThreadStart
thread caused non-unwinding panic. aborting.
error: process didn't exit successfully: `target\debug\psst-gui.exe` (exit code: 0xc0000409, STATUS_STACK_BUFFER_OVERRUN)
```

**Comment by jacksongoode** [comment]
Oh that's pretty odd. I wonder when this change took place... @SO9010 Do you have a Windows machine to test a fix?

**Comment by ericmedina024** [comment]
I'm happy to test if needed!

**Comment by SO9010** [comment]
I tested it, and @ericmedina024 is right: There is an error when the cache folder is not found. To get to the error, I had to delete that folder, which in macOS and Linux automatically gets regenerated.

**Comment by SO9010** [comment]
Strange as this has never been an issue before!

**Comment by SO9010** [comment]
Ok, so this is only an issue when using the prebuilt binary! So I am thinking it could be Microsoft AV!

**Comment by SO9010** [comment]
I only get this issue when using the prebuilt binaries. 

> ``` warning: error finalizing incremental compilation session directory `\\?\C:\Users\Eric\RustroverProjects\psst\target\debug\incremental\psst_protocol-268ihjyy4u29u\s-h1g26aime5-0otynen-working`: Access is denied. (os error 5) ```

Have you tried running Rust Rover as an administrator? Is your AV may be interfering? There is only two things that I can think of that is the issue and that is that PSST needs to be ran as administrator (which I have tried with no luck) and then Microsoft antivirus not liking the program and blocks it from creating new files?

**Comment by SO9010** [comment]
I'm a bit stumped by this! The code is all alright, @jacksongoode perhaps this comes from updating some of the dependencies? Just weird that it (for me at least) it is only an issue in the pre-built binary.

**Comment by ericmedina024** [comment]
> Have you tried running Rust Rover as an administrator? Is your AV may be interfering? There is only two things that I can think of that is the issue and that is that PSST needs to be ran as administrator (which I have tried with no luck) and then Microsoft antivirus not liking the program and blocks it from creating new files?

I just tried running `cargo run --bin psst-gui` from an admin command prompt and I receive the same warning:

**Comment by jacksongoode** [comment]
https://github.com/jpochyla/psst/pull/554

Could someone with Windows check out the artifacts from this build, I think it might have to do with a package that was changed.

**Comment by SO9010** [comment]
I think it best if @ericmedina024 tests it as I don't get the issue on my windows machine when I build it my self!

**Comment by uyen9vba** [comment]
I am having same issue.

**Comment by jacksongoode** [comment]
@uyen9vba Could you try running the artifact here https://github.com/jpochyla/psst/suites/30496192407/artifacts/2144991435

**Comment by ericmedina024** [comment]
> #554
> 
> Could someone with Windows check out the artifacts from this build, I think it might have to do with a package that was changed.

works on my machine!

**Comment by jacksongoode** [comment]
I believe this has been fixed in the latest build #554
